src/haxelib/api/Vcs.hx

```diff
@@ -347,6 +350,20 @@ class Git extends Vcs {
 		// return prev. cwd:
 		Sys.setCwd(oldCwd);
 	}
+
+	public function getReproducibleVersion(libPath: String): VcsData {
+		final oldCwd = Sys.getCwd();
+		Sys.setCwd(libPath);
+
+		final ret: VcsData = {
+			// could the remote's name not be "origin"?
```

Comment:
Well yeah it could.. but "most of the time it will work all the time" :shrug: 

Might need some error handling to recover just in case?